nf_core/synced_repo.py

```diff
@@ -314,15 +314,25 @@ def component_files_identical(self, component_name, base_path, commit, component
             self.checkout_branch()
         else:
             self.checkout(commit)
-        component_files = ["main.nf", "meta.yml"]
+        required_files = ["main.nf", "meta.yml"]
+        optional_files = ["environment.yml", "tests/main.nf.test", "tests/main.nf.test.snap", "tests/tags.yml"]
+        component_files = required_files + optional_files
         files_identical = {file: True for file in component_files}
         component_dir = self.get_component_dir(component_name, component_type)
         for file in component_files:
+            component_file = os.path.join(component_dir, file)
+            base_file = os.path.join(base_path, file)
+            if (file in optional_files) and (not os.path.exists(component_file)) and (not os.path.exists(base_file)):
+                log.debug(f'The optional file "{file}" was not present.')
+                # TO-DO: Not sure we need to log this, but if we do what should the msg be?
+                # In this case, `files_identical[file]` will remain `True`. Is that what we want?
+                continue
             try:
-                files_identical[file] = filecmp.cmp(os.path.join(component_dir, file), os.path.join(base_path, file))
+                files_identical[file] = filecmp.cmp(component_file, base_file)
             except FileNotFoundError:
-                log.debug(f"Could not open file: {os.path.join(component_dir, file)}")
-                continue
+                errmsg = f"Could not open file: {os.path.join(component_dir, file)}"
```

Comment:
`base_dir/path` could also not exist. better to take the filename from `FileNotFoundError`